### ISSUE TITLE
linuxPackages.nvidia_x11.settings is failing to build

### ISSUE THREAD

**Issue by NinjaTrappeur** [NixOS/nixpkgs#206213]
### Describe the bug
`linuxPackages.nvidia_x11.settings` is failing to build on the current master (7634c3250b6870e76f2e611cb95b72392add9aef). It seems like there's something off with the upstream makefile.

Trace:
```
make[1]: Entering directory '/build/source/src'
mkdir -p /nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/bin
install -m 755 _out/Linux_x86_64/nvidia-settings /nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/bin/nvidia-settings
mkdir -p /nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib
install -m 644 _out/Linux_x86_64/libnvidia-gtk2.so \
    /nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib/libnvidia-gtk2.so.525.60.11
install -m 644 _out/Linux_x86_64/libnvidia-gtk3.so \
    /nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib/libnvidia-gtk3.so.525.60.11
mkdir -p /nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib
install -m 644  \
    /nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib/
install: missing destination file operand after '/nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib/'
Try 'install --help' for more information.
make[1]: *** [Makefile:323: WAYLAND_LIB_install] Error 1
make[1]: Leaving directory '/build/source/src'
make: *** [Makefile:23: install] Error 2
error: builder for '/nix/store/b651iakjsc43n8j5j82dx628xcipc4lb-nvidia-settings-525.60.11.drv' failed with exit code 2;
       last 10 log lines:
       > install -m 644 _out/Linux_x86_64/libnvidia-gtk3.so \
       >     /nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib/libnvidia-gtk3.so.525.60.11
       > mkdir -p /nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib
       > install -m 644  \
       >     /nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib/
       > install: missing destination file operand after '/nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib/'
       > Try 'install --help' for more information.
       > make[1]: *** [Makefile:323: WAYLAND_LIB_install] Error 1
       > make[1]: Leaving directory '/build/source/src'
       > make: *** [Makefile:23: install] Error 2
       For full logs, run 'nix log /nix/store/b651iakjsc43n8j5j82dx628xcipc4lb-nvidia-settings-525.60.11.drv'.
```

Current workaround: 

```nix
  hardware.nvidia.nvidiaSettings = false;
```

:upside_down_face: 

### Steps To Reproduce
1. `nix-build -A linuxPackages.nvidia_x11.settings`

### Notify maintainers
@ncfavier @Kiskae

**Comment by Kiskae** [comment]
It looks like this happens if the Makefile fails on `pkg-config --exists wayland-client`.
The wayland library install target always gets called, but if wayland isn't detected it isn't initialized to an actual file.

However the derivation builds just fine on my end with the exact same buildpath, so I am not entirely sure what is causing this build failure. The wayland library is a transitive dependency through `gtk3`, so it should be present even if it isn't explicitly listed.

**Comment by ncfavier** [comment]
Same derivation builds fine here as well. @NinjaTrappeur are you building with the sandbox enabled? Could you try the flake interface? (`nix --extra-experimental-features 'nix-command flakes' build .#linuxPackages.nvidia_x11.settings`)

Would be nice to know why this is failing, but I guess in any case we can add `wayland` as an explicit dependency?

**Comment by NinjaTrappeur** [comment]
I am building the derivation with sandbox enabled.

I can reproduce with the flake interface:

```
~/code-root/github.com/NixOS/nixpkgs (master*) » nix build .\#linuxPackages.nvidia_x11.settings                   1 ↵ ninjatrappeur@trantor
warning: Git tree '/home/ninjatrappeur/code-root/github.com/NixOS/nixpkgs' is dirty
error: builder for '/nix/store/b651iakjsc43n8j5j82dx628xcipc4lb-nvidia-settings-525.60.11.drv' failed with exit code 2;
       last 10 log lines:
       > install -m 644 _out/Linux_x86_64/libnvidia-gtk3.so \
       >     /nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib/libnvidia-gtk3.so.525.60.11
       > mkdir -p /nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib
       > install -m 644  \
       >     /nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib/
       > install: missing destination file operand after '/nix/store/w68fkzs8q9bmi9gv36f264yb4h0ckps9-nvidia-settings-525.60.11/lib/'
       > Try 'install --help' for more information.
       > make[1]: *** [Makefile:323: WAYLAND_LIB_install] Error 1
       > make[1]: Leaving directory '/build/source/src'
       > make: *** [Makefile:23: install] Error 2
       For full logs, run 'nix log /nix/store/b651iakjsc43n8j5j82dx628xcipc4lb-nvidia-settings-525.60.11.drv'.
```

Are you sure you're building this on top of https://github.com/NixOS/nixpkgs/commit/7634c3250b6870e76f2e611cb95b72392add9aef?

Looking at the makefile, the wayland detection does not seem too crazy:

```


Looking at the derivation closure, it seems like wayland-client (coming from the nixpkgs wayland derivation default output) seems to be already a dependency of nvidia-x11.

![Capture d’écran du 2022-12-15 19-46-21](https://user-images.githubusercontent.com/1219785/207942460-2f7eba18-8c27-4180-9d7d-b1a7d04fb6fd.png)

**Comment by NinjaTrappeur** [comment]
Heeeh, I fat fingered that one and clicked on the close button by mistake ><

**Comment by NinjaTrappeur** [comment]
Wow, this one was wild.

My desktop has been behaving strangely for the last week: regularily corrupting my filesystems, crashing programs at random.

I realized this afternoon it was a hardware issue: a memory stick was faulty.

![Output of memcheck86 showing a memory stick is faulty](https://user-images.githubusercontent.com/1219785/207944677-4bf17bb9-d7dd-480f-8657-29c3ae6c5584.jpg)

I removed the faulty ram stick, things started to work again as expected.

While shuffling things around trying to fix this issue, I realized my nix store had some strange corruptions:

```
error: reading directory '/nix/store/REDACTED-source/pkgs/development/python-modules/flask-bcrypt': Bad message
```

I started to get suspicious about the store content, I decided to GC it. Aaaaand, after GC-ing it, I can't reproduce the issue anymore o0

I really don't understand what happened here… …fs corruption?

In any case: I can't reproduce the failure anymore. Closing the issue. Thanks for your investigations folks, I owe you one. Sorry for the noise!
